### PR TITLE
Fix #5998: Support fallback font correctly

### LINKED ISSUE #5998: \math??{} font styles are ignored in 2.x
The [STIX Fonts Demo](http://matplotlib.org/devdocs/examples/pylab_examples/stix_fonts_demo.html) shows three Blackboard entries with the same font. In [the 1.5.x line](http://matplotlib.org/examples/pylab_examples/stix_fonts_demo.html), these entries are italic, regular and bold. Additionally, the last three characters in the top row of circled numbers should be bold as well.

A bisect points to 4a2a18689794b6f0c6d35c55e4691e36bc47d5f8 as the first bad commit, the merging of #5214, so ping @zblz.

### LINKED ISSUE #6015: scatterplot axis autoscale fails for small data values
I'm finding that the scatterplot autoscaling fails when plotting small values, e.g. <0.001.

The autoscale axis range will never go below -0.008 to 0.008 for me, regardless of how small the range of the data being plotted.

```
plt.figure()
x = np.array([0,1,2,3])
x = np.array([2,4,5,9])
plt.scatter(x,y)
```

Data in a normal range produces
properly_scaled

```
plt.figure()
x = np.array([0,1,2,3])
x = np.array([2,4,5,9])
plt.scatter(x/10000,y/10000)
```

While scaling it down produces
badly_scaled

### PR DESCRIPTION
**Pull request by mdboom**
Cc: @zblz

### CODE REVIEW

File: lib/matplotlib/mathtext.py [hunk 1]
```diff
@@ -887,12 +887,12 @@ class DejaVuFonts(UnicodeFonts):
 
     def __init__(self, *args, **kwargs):
         # This must come first so the backend's owner is set correctly
+        TruetypeFonts.__init__(self, *args, **kwargs)
```
Comment (QuLogic) [inline lib/matplotlib/mathtext.py]: Is the comment above still true?

Comment (mdboom) [inline lib/matplotlib/mathtext.py]: Actually, I think it turned out that this move wasn't necessary after all -- I had meant to revert it before committing.  I'll do so now.

**Comment by zblz** [comment]
Looks good to me! :+1: 

Do we have any idea why forcing an italics class was initially needed? Or was it just a bug?

**Comment by mdboom** [comment]
In the context of STIX falling back to Bakoma, it needs to be hardcoded to `it` because the symbols the fallback was needed for are only in `it`.  I think this was copied-and-pasted to the context of DejaVu falling back to STIX, where that is not the case.

**Comment by QuLogic** [comment]
Are there no italic numbers? 05 vs 15?

I also wonder what is wrong with the baseline of the 't'.

**Comment by mdboom** [comment]
@QuLogic: That's correct.  There are no italic circled numbers.  Not sure what you mean by "05 vs. 15"

The baseline of t is a larger issue (addressed by #5414, but it will be some time to get that done).  But that's not introduced by this PR.

**Comment by QuLogic** [comment]
> @QuLogic: That's correct. There are no italic circled numbers. Not sure what you mean by "05 vs. 15"

No, I didn't mean circled, just plain italic numbers, e.g., `mathfont_dejavusans_05.png` vs. `mathfont_dejavusans_15.png` specify `mathrm` vs. `mathit`. But I see this is the case for the fonts as well.

**Comment by mdboom** [comment]
Yes -- there are no italic numbers in any of these fonts.

**Comment by mdboom** [comment]
This is ready to merge.

**Comment by tacaswell** [comment]
backported to v2.x as 819f746

**Pull request merged**